DEFEDIT-695 Ensure git user is configured

ragnard-king opened the pull request:
If user has not configured a git `user.name` and/or `user.email`, try to set them based on user info obtained when logging in.

- In editor1, we write user.name and user.email to local repo config when cloning. Can't do this in editor2 as repo location is determined by user and can be used with different logins at different times. Instead, we try to fallback to login information if no user settings have been configured.

Fixes https://github.com/defold/editor2-issues/issues/390, https://github.com/defold/editor2-issues/issues/12

ragnard-king committed:
DEFEDIT-695 Ensure git user is configured

matgis-king reviewed (changes requested):
Reviewed. An issue remains if the user have not logged out and gotten the new `"first-name"` and `"last-name"` prefs at login.

<img src="https://cloud.githubusercontent.com/assets/22448941/22327586/42968e64-e3b8-11e6-9d31-16e0d6980773.png" width="383">

ragnard-king committed:
DEFEDIT-695 Fixes after review

- git-name returns nil if both first-name and last-name are unset
- rename ChangesView git property for clarity

matgis-king reviewed (approved):
Looks good now!
